internal/oracles/mgr.go

```diff
@@ -50,18 +50,25 @@ func NewOracleMgr(config map[string]map[string]string, eventStore *events.EventS
 
 // Start starts the oracle manager.
 // It will block until Stop is called.
-func (omgr *OracleMgr) Start() {
+func (omgr *OracleMgr) Start() error {
 	ctx, cancel := context.WithCancel(context.Background()) // context that will be canceled when the manager shuts down
 	omgr.cancel = cancel
 	// Listen for status changes
 	// Start oracles if the node is a validator and is caught up
 	// Stop the oracles, if the node is not a validator
-	go func() {
-		// cancel function for the oracle instance
-		// if it is nil, then the oracle is not running
-		var oracleInstanceCancel context.CancelFunc
-
-		for {
+	// cancel function for the oracle instance
+	// if it is nil, then the oracle is not running
+	var oracleInstanceCancel context.CancelFunc
+
+	var errChan = make(chan error, 1)
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case err := <-errChan:
+			return err
+		case <-time.After(1 * time.Second):
 			// still in the catch up mode, do nothing
 			if omgr.cometNode.IsCatchup() {
 				continue
```

Comment:
To confirm, the `break` in the error condition after `GetValidators` is actually intended to break the `switch` , no the `for` loop, and is thus semantically equivalent to `continue`?

Reply:
I think it is intended to break out of the whole thing.